src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -143,10 +148,22 @@ pub struct Constraint {
     /// At this location.
     point: Location,
 
+    /// Later on, we thread the constraints onto a linked list
+    /// sorted by their `sub` field. So if you had:
```

Comment:
The use of the word "sort" in this comment confused me: "There's a single list  and its sorted by the `sub` field? Or there's a bunch of lists and each is individually sorted by the `sub` field -- but how is there any kind of sort  order here?"

I usually treat the term "sort" as meaning "make an ordered sequence"; but here you mean it more as "group together (via a linked list) all of the constraints that have the same `sub` field."